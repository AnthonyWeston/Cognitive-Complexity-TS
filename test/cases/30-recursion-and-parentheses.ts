function parens() {
    (parens)();
}

class C {
    bracketedThis() {
        (this).bracketedThis();
    }

    bracketedCallExpression() {
        (this.bracketedCallExpression)();
    }
}

const k = function l() {
    (l)();
}

class NoParentheses {
    constructor() {
        new NoParentheses;
    }
}

class Parentheses {
    constructor() {
        new (Parentheses);
    }
}

class BracketedThis {
    constructor() {
        (this).constructor();
    }
}

class BracketedConstructor {
    constructor() {
        (this.constructor)();
    }
}

const val = function func() {
    (val)();
}