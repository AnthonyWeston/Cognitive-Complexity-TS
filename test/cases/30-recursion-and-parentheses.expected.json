{
    "30-recursion-and-parentheses.ts": {
        "score": 7,
        "inner": [
            {
                "name": "parens",
                "score": 1,
                "line": 1,
                "column": 1
            },
            {
                "name": "C",
                "score": 2,
                "line": 5,
                "column": 1,
                "inner": [
                    {
                        "name": "bracketedThis",
                        "score": 1,
                        "line": 6,
                        "column": 5
                    },
                    {
                        "name": "bracketedCallExpression",
                        "score": 1,
                        "line": 10,
                        "column": 5
                    }
                ]
            },
            {
                "name": "l",
                "score": 0,
                "line": 15,
                "column": 11
            },
            {
                "name": "NoParentheses",
                "score": 1,
                "line": 19,
                "column": 1,
                "inner": [
                    {
                        "name": "constructor",
                        "score": 1,
                        "line": 20,
                        "column": 5
                    }
                ]
            },
            {
                "name": "Parentheses",
                "score": 1,
                "line": 25,
                "column": 1,
                "inner": [
                    {
                        "name": "constructor",
                        "score": 1,
                        "line": 26,
                        "column": 5
                    }
                ]
            },
            {
                "name": "BracketedThis",
                "score": 1,
                "line": 31,
                "column": 1,
                "inner": [
                    {
                        "name": "constructor",
                        "score": 1,
                        "line": 32,
                        "column": 5
                    }
                ]
            },
            {
                "name": "BracketedConstructor",
                "score": 1,
                "line": 37,
                "column": 1,
                "inner": [
                    {
                        "name": "constructor",
                        "score": 1,
                        "line": 38,
                        "column": 5
                    }
                ]
            },
            {
                "name": "func",
                "score": 1,
                "line": 43,
                "column": 13
            }
        ]
    }
}
